<!--
    <param name="maxUrange" value="8.0"/>
    <param name="maxRange" value="8.0"/>
-->
<launch>
  <arg name="scan_topic"  default="scan" />
  <arg name="base_frame"  default="base_footprint"/>
  <arg name="odom_frame"  default="odom"/>

  <arg name="bag_path" default="$(env HOME)/.ob/bag"/>
  <arg name="bag_name" default="ob_record_data"/>
  <!--<arg name="map_path" default="$(env HOME)/.ob/document/map"/>-->
  <!--<arg name="map_name" default="maptest"/>-->
  <arg name="map_name" default="$(env HOME)/.ob/map/ob_autoslam_offline"/>

  <node pkg="gmapping" type="slam_gmapping_replay" name="slam_gmapping_replay" output="screen" respawn="false" args="
  --scan_topic /scan
  --bag_filename $(arg bag_path)/$(arg bag_name).bag
  --on_done 'rosrun map_server map_saver -f $(arg map_name)'
  ">

    <param name="base_frame" value="$(arg base_frame)"/>
    <param name="odom_frame" value="$(arg odom_frame)"/>
    <param name="map_update_interval" value="1"/>
    <param name="maxUrange" value="8.0"/>
    <param name="maxRange" value="10.0"/>
    <param name="sigma" value="0.05"/>
    <param name="kernelSize" value="3"/>
    <param name="lstep" value="0.05"/>
    <param name="astep" value="0.05"/>
    <param name="iterations" value="5"/>
    <param name="lsigma" value="0.075"/>
    <param name="ogain" value="3.0"/>
    <param name="lskip" value="0"/>
    <param name="minimumScore" value="30"/> <!--30-->
    <param name="srr" value="0.01"/>
    <param name="srt" value="0.02"/>
    <param name="str" value="0.01"/>
    <param name="stt" value="0.02"/>
    <param name="linearUpdate" value="0.05"/>
    <param name="angularUpdate" value="0.0436"/>
    <param name="temporalUpdate" value="-1.0"/>
    <param name="resampleThreshold" value="0.5"/>
    <param name="particles" value="100"/><!--30-->
  <!--
    <param name="xmin" value="-50.0"/>
    <param name="ymin" value="-50.0"/>
    <param name="xmax" value="50.0"/>
    <param name="ymax" value="50.0"/>
  make the starting size small for the benefit of the Android client's memory...
  -->
    <param name="xmin" value="-1.0"/>
    <param name="ymin" value="-1.0"/>
    <param name="xmax" value="1.0"/>
    <param name="ymax" value="1.0"/>

    <param name="delta" value="0.05"/>
    <param name="llsamplerange" value="0.01"/>
    <param name="llsamplestep" value="0.01"/>
    <param name="lasamplerange" value="0.005"/>
    <param name="lasamplestep" value="0.005"/>
    <remap from="scan" to="$(arg scan_topic)"/>
  </node>
</launch>
